internal/net/http/transport/roundtrip.go

```diff
@@ -51,12 +51,26 @@ func (e *ert) RoundTrip(req *http.Request) (res *http.Response, err error) {
 		return e.roundTrip(req)
 	}
 
-	r, err := e.bo.Do(req.Context(), func() (interface{}, error) {
-		return e.roundTrip(req)
+	var boErr error
+	r, err := e.bo.Do(req.Context(), func() (res interface{}, err error) {
+		res, err = e.roundTrip(req)
+		if err != nil {
+			// if the error is retryable, return the error and let backoff to retry.
+			if errors.Is(err, errors.ErrTransportRetryable) {
+				return nil, err
+			}
+			// if the error is not retryable, return nil error to backoff to terminate the backoff execution
+			boErr = err
+			return nil, nil
+		}
+		return res, nil
 	})
 	if err != nil {
 		return nil, err
 	}
+	if boErr != nil {
```

Comment:
Thanks! it's better than before.
however the naming of this variable is a little bit strange. `boErr` is not an error of backoff.

Reply:
what do you think about this?
https://github.com/vdaas/vald/projects/3#card-42651099

Reply:
no.1 seems to be the better choice.
There are already several lines of codes that checks the returned error is retryable or not in agent-sidecar codes. so please refactor them at the same time.